metrics/web/test/dashboard/presentation/widgets/build_result_bar_component_test.dart

```diff
@@ -0,0 +1,396 @@
+      date: DateTime.now(),
+      duration: Duration.zero,
+      buildStatus: BuildStatus.successful,
+    );
+
+    BuildResultViewModel createBuildResult(BuildStatus status) {
+      return FinishedBuildResultViewModel(
+        buildResultPopupViewModel: popupViewModel,
+        date: DateTime.now(),
+        duration: const Duration(seconds: 20),
+        buildStatus: status,
+        url: 'url',
+      );
+    }
+
+    final successfulBuildResult = createBuildResult(BuildStatus.successful);
+    final failedBuildResult = createBuildResult(BuildStatus.failed);
+    final unknownBuildResult = createBuildResult(BuildStatus.unknown);
+    final inProgressBuildResult = InProgressBuildResultViewModel(
+      buildResultPopupViewModel: popupViewModel,
+      date: DateTime.now(),
+    );
+
+    final urlLauncherMock = _UrlLauncherMock();
+    UrlLauncherPlatform.instance = urlLauncherMock;
+
+    tearDown(() {
+      reset(urlLauncherMock);
+    });
+
+    testWidgets(
+      "displays the PlaceholderBar if the given build result is null",
+      (tester) async {
+        await tester.pumpWidget(
+          const _BuildResultBarComponentTestbed(
+            buildResult: null,
+          ),
+        );
+
+        expect(placeholderFinder, findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "applies the bar color from the inactive widget theme to the PlaceholderBar",
+      (WidgetTester tester) async {
+        final expectedColor = metricsTheme.inactiveWidgetTheme.primaryColor;
+
+        await tester.pumpWidget(
+          const _BuildResultBarComponentTestbed(
+            buildResult: null,
+            themeData: metricsTheme,
+          ),
+        );
+
+        final placeholderBar = tester.widget<PlaceholderBar>(placeholderFinder);
+
+        expect(placeholderBar.color, expectedColor);
+      },
+    );
+
+    testWidgets(
+      "applies the bar width from the dimension config to the PlaceholderBar",
+      (WidgetTester tester) async {
+        const expectedWidth = DimensionsConfig.graphBarWidth;
+
+        await tester.pumpWidget(
+          const _BuildResultBarComponentTestbed(
+            buildResult: null,
+          ),
+        );
+
+        final placeholderBar = tester.widget<PlaceholderBar>(placeholderFinder);
+
+        expect(placeholderBar.width, expectedWidth);
+      },
+    );
+
+    testWidgets(
+      "displays the build result bar with the given build result",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          _BuildResultBarComponentTestbed(
+            buildResult: successfulBuildResult,
+          ),
+        );
+
+        final buildResultBarFinder = find.byWidgetPredicate((widget) {
+          return widget is BuildResultBar &&
+              widget.buildResult == successfulBuildResult;
+        });
+
+        expect(buildResultBarFinder, findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the build result bar in a stack with a passthrough fit",
```

Comment:
```suggestion
      "displays the build result bar in a stack with a pass-through fit",
```
Also, I believe it would be great to test that this `Stack` has a  `Clip.none` clip behavior since it seems like it is a pretty critical part of this widget. What do you think?